osu.Game/Overlays/Settings/Sections/Maintenance/MigrationRunScreen.cs

```diff
@@ -0,0 +1,115 @@
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Screens;
+using osuTK;
+
+namespace osu.Game.Overlays.Settings.Sections.Maintenance
+{
+    public class MigrationRunScreen : OsuScreen
+    {
+        private readonly DirectoryInfo destination;
+
+        [Resolved(canBeNull: true)]
+        private OsuGame game { get; set; }
+
+        public override bool AllowBackButton => false;
+
+        public override bool AllowExternalScreenChange => false;
+
+        public override bool DisallowExternalBeatmapRulesetChanges => true;
+
+        public override bool HideOverlaysOnEnter => true;
+
+        private Task migrationTask;
+
+        public MigrationRunScreen(DirectoryInfo destination)
+        {
+            this.destination = destination;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            InternalChildren = new Drawable[]
+            {
+                new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Direction = FillDirection.Vertical,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Spacing = new Vector2(10),
+                    Children = new Drawable[]
+                    {
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Text = "Migration in progress",
+                            Font = OsuFont.Default.With(size: 40)
+                        },
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Text = "This could take a few minutes depending on the speed of your disk(s).",
+                            Font = OsuFont.Default.With(size: 30)
+                        },
+                        new LoadingSpinner(true)
+                        {
+                            State = { Value = Visibility.Visible }
+                        },
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Text = "Please avoid interacting with the game!",
+                            Font = OsuFont.Default.With(size: 30)
+                        },
+                    }
+                },
+            };
+
+            Beatmap.Value = Beatmap.Default;
+
+            migrationTask = Task.Run(PerformMigration)
+                                .ContinueWith(t =>
+                                {
+                                    if (t.IsFaulted)
+                                        Logger.Log($"Error during migration: {t.Exception?.Message}", level: LogLevel.Error);
```

Comment:
Arguably should be

```suggestion
                                        Logger.Error(t.Exception, "Error during migration: {t.Exception?.Message}");
```

as the current version completely discards stack traces, which means we won't be able to debug user reports in any of this.

Reply:
Using `Logger.Error` actually fires the error to our server handler right now, so we avoid using it in user scenarios like this. Maybe not the best handling there.